Objects/abstract.c

```diff
@@ -1714,6 +1715,61 @@ PyNumber_ToBase(PyObject *n, int base)
     return res;
 }
 
+typedef struct {
+    const uint16_t slot;
+    const uint16_t islot;
+} nb_slot_info;
+
+#define NB_SLOT_INFO(slot) {NB_SLOT(nb_##slot), NB_SLOT(nb_inplace_##slot)}
+
+static const nb_slot_info nb_slot_infos[] = {
+    [NB_AND] = NB_SLOT_INFO(and),
+    [NB_FLOOR_DIVIDE] = NB_SLOT_INFO(floor_divide),
+    [NB_LSHIFT] = NB_SLOT_INFO(lshift),
+    [NB_MATRIX_MULTIPLY] = NB_SLOT_INFO(matrix_multiply),
+    [NB_OR] = NB_SLOT_INFO(or),
+    [NB_RSHIFT] = NB_SLOT_INFO(rshift),
+    [NB_SUBTRACT] = NB_SLOT_INFO(subtract),
+    [NB_TRUE_DIVIDE] = NB_SLOT_INFO(true_divide),
+    [NB_XOR] = NB_SLOT_INFO(xor),
+};
+
+#undef NB_SLOT_INFO
+
+typedef struct {
+    const char name[3];
+    const char iname[4];
+} nb_name_info;
+
+#define NB_NAME_INFO(name) {name, name "="}
+
+static const nb_name_info nb_name_infos[] = {
+    [NB_AND] = NB_NAME_INFO("&"),
+    [NB_FLOOR_DIVIDE] = NB_NAME_INFO("//"),
+    [NB_LSHIFT] = NB_NAME_INFO("<<"),
+    [NB_MATRIX_MULTIPLY] = NB_NAME_INFO("@"),
+    [NB_OR] = NB_NAME_INFO("|"),
+    [NB_RSHIFT] = NB_NAME_INFO(">>"),
+    [NB_SUBTRACT] = NB_NAME_INFO("-"),
+    [NB_TRUE_DIVIDE] = NB_NAME_INFO("/"),
+    [NB_XOR] = NB_NAME_INFO("^"),
+};
+
+#undef NB_NAME_INFO
+
+PyObject *
+_PyNumber_Op(PyObject *o1, PyObject *o2, unsigned op)
```

Comment:
This just forwards to `binary_op`. You might want to consider:
1. Make this the function that does the work (effectively inlining `binary_op` here).
2. Make the old API `PyNumber_TrueDivide`, etc. call `_PyNumber_Op`
3. Delete `binary_op`.